modules/infra/api/gr_infra.h

```diff
@@ -39,14 +39,23 @@
 
 #define GR_VRF_ID_ALL UINT16_MAX
 
+enum gr_iface_mode {
+	GR_IFACE_MODE_L3,
+	GR_IFACE_MODE_L2_XCONNECT,
+};
+
 // Generic struct for all network interfaces.
 struct gr_iface {
 	uint16_t id; // Interface unique index.
 	uint16_t type; // Interface type. Uses values from GR_IFACE_TYPE_*.
 	uint16_t flags; // Interface flags. Bit mask of GR_IFACE_F_*.
 	uint16_t state; // Interface state. Bit mask of GR_IFACE_S_*.
 	uint16_t mtu; // Maximum transmission unit size (incl. headers).
-	uint16_t vrf_id; // L3 addressing and routing domain
+	union {
+		uint16_t vrf_id; // L3 addressing and routing domain
+		uint16_t bridge_domain; // L2 xconnect peer interface id
+	};
+	enum gr_iface_mode mode;
```

Comment:
If you use enums in structures shared over the API, could you either pack them (or at least store them in well-known integer types)?

Reply:
Also, move the mode field before vrf_id, domain_id?